backend/src/routes/auth/oauth.rs

```diff
@@ -46,23 +53,41 @@ pub async fn github_callback(
         .map_err(|_| Error::OAuth)?;
 
     let token = format!("Bearer {}", token_result.access_token().secret());
-    let response = reqwest::Client::new()
+    let response_user = reqwest::Client::new()
         .get("https://api.github.com/user")
-        .header("Authorization", token)
+        .header("Authorization", &token)
         .header("User-Agent", "hack4krak-backend")
         .send()
         .await?;
 
-    if !response.status().is_success() {
+    if !response_user.status().is_success() {
         return Err(Error::InvalidCredentials);
     }
 
-    let user: GitHubUser = response
-        .json()
-        .await
-        .map_err(|_| Error::InvalidCredentials)?;
+    let mut user: GitHubUser = response_user.json().await.map_err(|_| InvalidCredentials)?;
```

Comment:
Return different error, it will be eaisier to find those bugs in future. We should return `500` or something like that